packages/openapi-v3/src/controller-spec.ts

```diff
@@ -248,24 +251,50 @@ function processSchemaExtensions(
   assignRelatedSchemas(spec, schema.definitions);
   delete schema.definitions;
 
-  if (isReferenceObject(schema)) return;
+  /**
+   * check if we have been provided a `not`
+   * `not` is valid in many cases- here we're checking for
+   * `not: { schema: {'x-ts-type': SomeModel }}
+   */
+  if (schema.not) {
+    processSchemaExtensions(spec, schema.not);
+  }
+
+  /**
+   *  check for schema.allOf, schema.oneOf, schema.anyOf arrays first.
+   *  You cannot provide BOTH a defnintion AND one of these keywords.
+   */
+  const hasOwn = (prop: string) =>
+    Object.prototype.hasOwnProperty.call(schema, prop);
```

Comment:
Can it be just the following?
```ts
const hasOwn = (prop: string) => schema.hasOwnProperty(prop);
```

Reply:
Sure!  Sorry about that, I'm so used to that being a lint-error/typescript-warning these days.